src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,77 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../types/Comment';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onAdd: (val: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onAdd, selectedPost }) => {
+  const [isSubmiting, setIsSubmiting] = useState(false);
+
+  const [name, setName] = useState('');
+  const [isErrorName, setIsErrorName] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [isErrorEmail, setIsErrorEmail] = useState(false);
+
+  const [comment, setComment] = useState('');
+  const [isErrorComment, setIsErrorComment] = useState(false);
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setIsErrorName(false);
+  };
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setIsErrorEmail(false);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setComment(e.target.value);
+    setIsErrorComment(false);
+  };
+
+  const handleReset = () => {
+    setComment('');
+    setEmail('');
+    setName('');
+    setIsErrorComment(false);
+    setIsErrorEmail(false);
+    setIsErrorName(false);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
```

Comment:
You should consider using `try {..} catch` for error handling in the `handleSubmit` function. This would allow you to handle any errors that might occur during the execution of the function.